classes/question/bank/difficulty_level_column.php

```diff
@@ -44,20 +44,60 @@ public function get_name() {
     }
 
     /**
+     * Set conditions to apply to join.
+     * @param  array $joinconditions Conditions to apply to join (via WHERE clause)
+     */
+    public function set_joinconditions($joinconditions) {
+        $this->joinconditions = $joinconditions;
+    }
+
+    /**
+     * Get params that this join requires be added to the query.
+     * @return array sqlparams required to be added to query
+     */
+    public function get_sqlparams() {
+        return $this->sqlparams;
+    }
+
+    /**
      * Get sql query join for this column
      * @return array sql query join additional
      */
     public function get_extra_joins() {
+        $tests = array('q.parent = 0');
+        $correcttests = array('rightanswer = responsesummary');
+        $totaltests = array('responsesummary IS NOT NULL');
+        // Have to build params array while building query, due to inability to use single
+        // named parameter multiple times. Each call to joincondition->where updates
+        // joincondition's params.
+        $this->sqlparams = array();
+        foreach ($this->joinconditions as $joincondition) {
+            if ($joincondition->where()) {
+                $tests[] = '((' . $joincondition->where() .'))';
+                $this->sqlparams = array_merge($this->sqlparams, $joincondition->params());
+                $correcttests[] = '((' . $joincondition->where() .'))';
+                $this->sqlparams = array_merge($this->sqlparams, $joincondition->params());
+                $totaltests[] = '((' . $joincondition->where() .'))';
+                $this->sqlparams = array_merge($this->sqlparams, $joincondition->params());
```

Comment:
I don't fully understand the difference between `$tests`, `$correcttests` and `$totaltests`, can you quickly elaborate on this. If you think it's worth it, maybe even add a comment to the code.